src/core/tetris.ts

```diff
@@ -69,296 +94,158 @@ class Tetris {
   }
 
   /**
-   * @brief: handleBoardUpdate: Handler for each 'tick' of the game
+   * @brief: updateGameStates: Handler for each 'tick' of the game
    * or when a command is issued
    * @param: command - The command to be executed. The command is
-   * always 'Down' for each tick of the game
+   * always 'Down' for each tick of the game. If the command is undefined,
+   * we simply return the current game states without making any changes
+   * @return: Updated game states
    */
-  public handleBoardUpdate(
-    command: TetrisConsts.Command = TetrisConsts.Command.Down
-  ): {
-    corX: number;
-    corY: number;
-    ghostCorY: number;
-    heldTetromino: TetrisConsts.Tetromino;
-    activeTetromino: TetrisConsts.Tetromino;
-    activeTetrominoRotate: TetrisConsts.Rotation;
-    field: TetrisCol[];
-    spawnedTetrominos: TetrisConsts.Tetromino[];
-    gameOver: boolean;
-    score: number;
-    level: number;
-    gameInterval: number;
-  } {
-    /* Handling init - We only render the newly spawned tetromino */
-    if (this.initRender) {
-      this.renderTetromino(
-        this.corX,
-        this.ghostCorY,
-        this.activeTetromino,
-        this.activeTetrominoRotate,
-        TetrisConsts.GHOST_TETROMINO_INDEX
-      );
-      this.renderTetromino(
-        this.corX,
-        this.corY,
-        this.activeTetromino,
-        this.activeTetrominoRotate,
-        this.activeTetromino
-      );
-
-      this.initRender = false;
-    } else {
-      /**
-       * Remove current tetromino from field for next logic by rendering
-       * blank pixels onto the board
-       */
-      this.renderTetromino(
-        this.corX,
-        this.ghostCorY,
-        this.activeTetromino,
-        this.activeTetrominoRotate,
-        TetrisConsts.Tetromino.Blank
-      );
-      this.renderTetromino(
-        this.corX,
-        this.corY,
-        this.activeTetromino,
-        this.activeTetrominoRotate,
-        TetrisConsts.Tetromino.Blank
-      );
-
-      let yAddValid = true;
-      let testCorX = 0;
-      let testCorY = 0;
-      let testRotate = 0;
-      let testOffset = [];
-
-      /* Determine which value to be modified (x - y - rotate ?) */
-      switch (command) {
-        case TetrisConsts.Command.Left: {
-          testCorX = this.corX - 1;
-
-          if (
-            this.isMoveValid(
-              testCorX,
-              this.corY,
-              this.activeTetromino,
-              this.activeTetrominoRotate
-            )
-          ) {
-            this.corX = testCorX;
-          }
-
-          break;
-        }
-        case TetrisConsts.Command.Right: {
-          testCorX = this.corX + 1;
-
-          if (
-            this.isMoveValid(
-              testCorX,
-              this.corY,
-              this.activeTetromino,
-              this.activeTetrominoRotate
-            )
-          ) {
-            this.corX = testCorX;
+  public updateGameStates(
+    command: Command | undefined): TetrisStates {
```

Comment:
I have a few nitpicks here. I think we should use `null` over `undefined ` because `null` represents the intentional absence of any object value, while  `undefined `  is just a variable that has not been assigned a value. We should probably pass `null ` as the default argument here as well.

`public updateGameStates(command: Command | null = null): TetrisStates { }`